src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageHandler.java

```diff
@@ -103,9 +104,9 @@ public void sendErrorResponse(HttpServletResponse response, CodedException ex) t
         }
         response.setCharacterEncoding(MimeUtils.UTF8);
         response.setHeader("X-Road-Error", ex.getFaultCode());
-        if (requestAcceptType.equalsIgnoreCase("text/xml")
-                || requestAcceptType.equalsIgnoreCase("application/xml")) {
-            response.setContentType(requestAcceptType);
+        if (StringUtils.containsIgnoreCase(requestAcceptType, "text/xml")
+                || StringUtils.containsIgnoreCase(requestAcceptType, "application/xml")) {
```

Comment:
This partially works, but in addition to (application|text)/xml it accepts any content type containing that string, e.g. `text/xml-patch+xml` or `application/xml-patch+xml` (https://www.iana.org/assignments/media-types/media-types.xhtml).

In addition, one could expect that `Accept: text/xml` would return the XML representation using the same content type, and `Accept: text/*` would also return the XML representation (`text/xml`) and not `application/json`.